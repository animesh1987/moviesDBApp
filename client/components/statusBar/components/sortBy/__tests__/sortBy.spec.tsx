import * as React from 'react';
import { create } from 'react-test-renderer';
import {shallow} from "enzyme";

import { SortBy } from '../index';

describe('Sort By Test', () => {

  const sortBy = jest.fn();

  it('renders correctly', () => {
    const component = create(
      <SortBy
        sortBy={sortBy} selected={'status'}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Sort By function works correctly', () => {
    const component = shallow(
      <SortBy
        sortBy={sortBy} selected={'rating'}/>);
    component.childAt(1).simulate('click');
    expect(sortBy).toBeCalled();
    expect(
      component.childAt(2).hasClass('active')).toBeTruthy();
    component.childAt(2).simulate('click');
    expect(sortBy).toBeCalled();
  });

  it('Active class applied correctly', () => {
    const component = shallow(
      <SortBy
        sortBy={sortBy} selected={'release_date'}/>);
    expect(
      component.childAt(1).hasClass('active')).toBeTruthy();
  });
});