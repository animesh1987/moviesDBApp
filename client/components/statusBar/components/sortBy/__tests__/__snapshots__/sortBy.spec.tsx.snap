// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort By Test renders correctly 1`] = `
<div>
  <span>
    Sort By
  </span>
  <span
    className="cursor-pointer "
    onClick={[Function]}
  >
    release date
  </span>
  <span
    className="cursor-pointer "
    onClick={[Function]}
  >
    rating
  </span>
</div>
`;
